onnx/defs/shape_inference.cc

```diff
@@ -230,6 +230,22 @@ void UnionShapeInfo(const TensorShapeProto& source_shape, TypeProto_Tensor& targ
   UnionShapeInfoForTensor(source_shape, target_type);
 }
 
+void UnionShapeInfo(const TypeProto_Tensor& source_type, TypeProto_Tensor& target_type) {
+  if (source_type.has_shape()) {
```

Comment:
Please explain the reason behind the code. E.g.,
```
We merge shapes from source and target if at least one of them exists. Otherwise, union with empty shape should remain empty shape.

Reply:
Done, thanks!